File: python/examples/staking.py
#!/usr/bin/env python3
"""
Staking Example

Stake and unstake HYPE tokens.

Requirements:
    pip install hyperliquid-sdk

Usage:
    export PRIVATE_KEY="0x..."
    python staking.py
"""

import os
import sys

from hyperliquid_sdk import HyperliquidSDK

PRIVATE_KEY = os.environ.get("PRIVATE_KEY")

if not PRIVATE_KEY:
    print("Error: Set PRIVATE_KEY environment variable")
    print("  export PRIVATE_KEY='0xYourPrivateKey'")
    sys.exit(1)


def main():
    print("Hyperliquid Staking Example")
    print("=" * 50)

    sdk = HyperliquidSDK(private_key=PRIVATE_KEY)
    print(f"Wallet: {sdk.address}")

    # Stake HYPE tokens
    # result = sdk.stake(100)
    # print(f"Stake: {result}")

    # Unstake HYPE tokens
    # result = sdk.unstake(50)
    # print(f"Unstake: {result}")

    # Delegate to a validator
    # result = sdk.delegate("0x...", 100)
    # print(f"Delegate: {result}")

    # Undelegate from a validator
    # result = sdk.undelegate("0x...", 50)
    # print(f"Undelegate: {result}")

    print()
    print("Staking methods available:")
    print("  sdk.stake(amount)")
    print("  sdk.unstake(amount)")
    print("  sdk.delegate(validator, amount)")
    print("  sdk.undelegate(validator, amount)")


if __name__ == "__main__":
    main()

File: typescript/examples/staking.ts
#!/usr/bin/env npx ts-node
// @ts-nocheck
/**
 * Staking Example
 *
 * Stake and unstake HYPE tokens.
 *
 * Requires: PRIVATE_KEY environment variable
 */

import { HyperliquidSDK } from '@quicknode/hyperliquid-sdk';

const PRIVATE_KEY = process.env.PRIVATE_KEY;

if (!PRIVATE_KEY) {
  console.error("Set PRIVATE_KEY environment variable");
  console.error("Example: export PRIVATE_KEY='0x...'");
  process.exit(1);
}

async function main() {
  const sdk = new HyperliquidSDK(undefined, { privateKey: PRIVATE_KEY });

  // Stake HYPE tokens
  // const result = await sdk.stake(100);
  // console.log(`Stake: ${JSON.stringify(result)}`);

  // Unstake HYPE tokens
  // const result = await sdk.unstake(50);
  // console.log(`Unstake: ${JSON.stringify(result)}`);

  // Delegate to a validator
  // const result = await sdk.delegate("0x...", 100);
  // console.log(`Delegate: ${JSON.stringify(result)}`);

  // Undelegate from a validator
  // const result = await sdk.undelegate("0x...", 50);
  // console.log(`Undelegate: ${JSON.stringify(result)}`);

  console.log("Staking methods available:");
  console.log("  sdk.stake(amount)");
  console.log("  sdk.unstake(amount)");
  console.log("  sdk.delegate(validator, amount)");
  console.log("  sdk.undelegate(validator, amount)");
}

main().catch(console.error);
